### ISSUE TITLE
JAB: nvda sometimes incorrectly adds "alt" modifiers to menu items

### ISSUE THREAD

**Issue by rpaquay** [nvaccess/nvda#6083]
Repo steps:

* Install Android Studio 2.2 Preview 3 (or later)
* Run Android Studio and nvda
* Create an empty project (or open an existing project) to get to the main IDE window
* Select the "Edit > Find" menu item

Result:
* nvda speaks "Find menu selected Alt+F"

Expected:
* nvda speaks "Find menu selected F"

The problem is that there is an incorrect assumption in the source code about top level menu items:
https://github.com/nvaccess/nvda/blob/1aae7484bd84b2f5757e8ee65911db9fd27b56fc/source/NVDAObjects/JAB/__init__.py#L248

> We assume alt  if there are no modifiers at all and its not a menu item as this is clearly a nmonic

I think the assumption is only correct is the parent of the item with the "menu" role has the "menubar" role.

See AccessibleRole definitions:

"menubar" role:
> An object usually drawn at the top of the primary dialog box of an application that contains a list of menus the user can choose from.  For example, a menu bar might contain menus for "File," "Edit," and "Help."

"menu" role:
> An object usually found inside a menu bar that contains a list of actions the user can choose from.  A menu can have any object as its children, but most often they are menu items, other menus, or rudimentary objects such as radio buttons, check boxes, or separators.  For example, an application may have an "Edit" menu that contains menu items for "Cut" and "Paste."

So, basically, when a "menu" is the direct child of a "menubar", it is a top level menu (e.g. "File). However, when  a "menu" is the child of another "menu", it is just a nested menu (e.g. "Edit").

It is a bit ridiculous that JAB does not return the correct "Alt" modifier to top level menu items and that screen readers have to implement such heuristics, but it is clearly too late to change that.

I propose a fix where the "alt" modifiers is added only when the parent has the "menubar" role

**Comment by leonardder** [comment]
There's a pr for this, I suggest a p3 to review it.

**Comment by Adriani90** [comment]
cc: @michaelDCurran
